### ISSUE TITLE
Fish shell closes immediately without error Cygwin

### ISSUE THREAD

**Issue by NickG123** [fish-shell/fish-shell#7336]
I tried to install fish on Cygwin for the first time on this machine, and trying to start the shell results in it closing immediately with no error code or message.  Any help is greatly appreciated, big fan of the project 😄 
Let me know if you need any more info.
Thanks,
Nick

System information
```
$ fish --version
fish, version 3.1.2

$ uname -a
CYGWIN_NT-10.0 LT-2029 3.1.7(0.340/5/3) 2020-08-22 17:48 x86_64 Cygwin
```

Attempting to run fish under Cygwin results in fish exiting immediately without an error code:
```
$ fish
$ echo $?
0
```

Running `sh -c 'env HOME=$(mktemp -d) fish'` does not change the behaviour

Debug output:
```
$ fish -d5
<2> fish: exec_path: '/usr/bin/fish', argv[0]: 'fish'
<2> fish: Using compiled in paths:
<2> fish: determine_config_directory_paths() results:
paths.data: /usr/share/fish
paths.sysconf: /etc/fish
paths.doc: /usr/share/doc/fish
paths.bin: /usr/bin
<2> fish: curses var TERM='xterm'
<2> fish: curses var TERMINFO missing or empty
<2> fish: curses var TERMINFO_DIRS missing or empty
<2> fish: 256 color support enabled for TERM=xterm
<2> fish: default emoji width: 1
<5> fish: universal log reading from file
<5> fish: intern [
<5> fish: intern and
<5> fish: intern argparse
<5> fish: intern begin
<5> fish: intern bg
<5> fish: intern bind
<5> fish: intern block
<5> fish: intern break
<5> fish: intern breakpoint
<5> fish: intern builtin
<5> fish: intern case
<5> fish: intern cd
<5> fish: intern command
<5> fish: intern commandline
<5> fish: intern complete
<5> fish: intern contains
<5> fish: intern continue
<5> fish: intern count
<5> fish: intern disown
<5> fish: intern echo
<5> fish: intern else
<5> fish: intern emit
<5> fish: intern end
<5> fish: intern eval
<5> fish: intern exec
<5> fish: intern exit
<5> fish: intern false
<5> fish: intern fg
<5> fish: intern for
<5> fish: intern function
<5> fish: intern functions
<5> fish: intern history
<5> fish: intern if
<5> fish: intern jobs
<5> fish: intern math
<5> fish: intern not
<5> fish: intern or
<5> fish: intern printf
<5> fish: intern pwd
<5> fish: intern random
<5> fish: intern read
<5> fish: intern realpath
<5> fish: intern return
<5> fish: intern set
<5> fish: intern set_color
<5> fish: intern source
<5> fish: intern status
<5> fish: intern string
<5> fish: intern switch
<5> fish: intern test
<5> fish: intern time
<5> fish: intern true
<5> fish: intern ulimit
<5> fish: intern wait
<5> fish: intern while
<2> fish: sourcing /usr/share/fish/config.fish
<5> fish: intern /usr/share/fish/config.fish
<5> fish: intern /usr/share/fish/config.fish
<5> fish: intern /usr/share/fish/config.fish
<5> fish: intern /usr/share/fish/config.fish
<5> fish: intern /usr/share/fish/config.fish
<5> fish: intern /usr/share/fish/config.fish
<5> fish: intern /usr/share/fish/__fish_build_paths.fish
<5> fish: intern /usr/share/fish/config.fish
<5> fish: intern /usr/share/fish/__fish_build_paths.fish
<5> fish: intern /usr/share/fish/config.fish
<5> fish: intern /usr/share/fish/config.fish
<5> fish: intern /usr/share/fish/config.fish
<5> fish: intern /usr/share/fish/config.fish
<5> fish: intern /usr/share/fish/config.fish
<5> fish: intern /usr/share/fish/config.fish
<5> fish: intern /usr/share/fish/config.fish
<5> fish: intern /usr/share/fish/config.fish
<5> fish: intern /usr/share/fish/config.fish
<5> fish: intern /usr/share/fish/config.fish
<5> fish: intern /usr/share/fish/config.fish
```

**Comment by faho** [comment]
As a guess, try `fish - i`

**Comment by NickG123** [comment]
No difference with `fish - i` (or `fish -i`).

Interestingly, even `fish --help` also exits without printing anything, though `fish --version` works

**Comment by faho** [comment]
So `fish --version` just directly prints the version, `fish --help` needs to do some magic to bring the manpage into printable shape.

This means:

- `--help` does much more
- `--help` won't work if you don't have the manpages installed

So if you do have the manpages installed, that means that it's not getting to the actually-running-commands part.

That makes it look like it's not an issue with reader setup, like I assumed at first.

Unfortunately I don't have a cygwin system to test and I don't have any real idea what's wrong here. The best I can do is to ask you to build fish from master, to see if that's fixed it.

**Comment by NickG123** [comment]
The man pages are installed, and `man fish` works just fine.
I probably won't have the chance to try building from source today, but I'll give it a shot when I can and let you know 😄 
Thanks for the quick responses.

**Comment by NickG123** [comment]
So... this is a strange one

Building master branch made no difference, I still got the same results.  I played around with the code a little bit trying to track the source of the error down.  
I had narrowed it down to stopping during the `parser.eval(cmd, io_chain_t());` line in `source_config_in_directory`...

And then, it started working, with seemingly no changes from me.
Not only that, the previously installed version from binary started working.

So clearly something strange is happening with my environment that's probably unrelated to `fish`.
I think (unless you have other recommendations, or tests you want me to try) it's probably safe to close this ticket.
If I run into trouble again and figure out what the problem is, I'll make sure to update the ticket.
